newrelic/hooks/hybridagent_opentelemetry.py

```diff
@@ -112,14 +155,26 @@ def set_attributes(self, attributes):
             for key, value in attributes.items():
                 add_custom_span_attribute(key, value)
 
-    otel_tracestate_headers = None
-
+    if transaction.settings.distributed_tracing.enabled:
+        if not isinstance(span, otel_api_trace.NonRecordingSpan):
+            # Use the Otel trace and span ids if current span
+            # is not a NonRecordingSpan.  Otherwise, we need to 
+            # override the current span with the transaction
+            # and trace guids.
```

Comment:
Probably be a good idea to cash this to save on compute time.
```suggestion
            # and trace guids.
            trace_id = span.get_span_context().trace_id
```